### PR TITLE
wandb.watch

### PR DESCRIPTION
**Pull request by Linux-cpp-lisp**
tested with real wandb account, `example.yaml`, and a restart

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [X] My code follows the code style of this project and has been formatted using `black`.
- [ ] All new and existing tests passed, including on GPU (if relevant).
- [X] I have added tests that cover my changes (if relevant).
- [X] The option documentation (`docs/options`) has been updated with new or changed options.
- [X] I have updated `CHANGELOG.md`.
- [X] I have updated the documentation (if relevant).

### CODE REVIEW

File: nequip/train/trainer_wandb.py [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
+        if not self._initialized:
+            return
 
-        # TODO: test and re-enable this
-        # wandb.watch(self.model)
+        wandb_watch = self.kwargs.get("wandb_watch", False)
+        if wandb_watch is not False:
+            if wandb_watch is True:
+                wandb_watch = {}
+            wandb.watch(self.model, **wandb_watch)
```
Comment (nw13slx) [inline nequip/train/trainer_wandb.py:23]:
```suggestion
        if self.kwargs.get("wandb_watch", False):
            wandb_watch_kwargs = self.kwargs.get("wandb_watch_kwargs", {})
            wandb.watch(self.model, **wandb_watch_kwargs)
```

Let's keep the type straight.

File: configs/full.yaml [hunk 2]
```diff
@@ -131,6 +131,12 @@ chemical_symbols:
 # logging
 wandb: false                                                                       # we recommend using wandb for logging, we'll turn it off here as it's optional
 wandb_project: toluene-example                                                     # project name used in wandb
+wandb_watch: false
+# see https://docs.wandb.ai/ref/python/watch
+# wandb_watch:
```
Comment (nw13slx) [inline configs/full.yaml:136]:
```suggestion
# wandb_watch_kwargs:
```

File: nequip/train/trainer_wandb.py [hunk 3]
```diff
@@ -1,31 +1,23 @@
...
-        Trainer.__init__(self, **kwargs)
+    """Trainer class that adds WandB features"""
 
     def end_of_epoch_log(self):
         Trainer.end_of_epoch_log(self)
         wandb.log(self.mae_dict)
 
-    def init_model(self):
+    def init(self):
+        super().init()
```
Comment (nw13slx) [inline nequip/train/trainer_wandb.py:14]:
```suggestion
        _initialized = self._initialized
        super().init()
```

File: nequip/train/trainer_wandb.py [hunk 4]
```diff
@@ -1,31 +1,23 @@
...
 
-        Trainer.init_model(self)
+        if not self._initialized:
```
Comment (nw13slx) [inline nequip/train/trainer_wandb.py:16]:
```suggestion
        if self._initialized:
```

File: nequip/train/trainer_wandb.py (hunk 3 above)
Comment (Linux-cpp-lisp) [reply nequip/train/trainer_wandb.py:14]:
```
def init(self):
        """initialize optimizer"""
        if self.model is None:
            return

        self.model.to(self.torch_device)
        self.init_objects()

        self._initialized = True
```
no? not if model is none which is what I'm guarding against

Comment (nw13slx) [reply nequip/train/trainer_wandb.py:14]: ah. you are right. I was confused.

**Pull request merged**